.github/workflows/deployment.yml

```diff
@@ -23,104 +43,144 @@ jobs:
            jdk14Path: /jdk-14.jdk/Contents/Home
-           archivePortable: tar -czf build/distribution/JabRef-portable_macos.tar.gz -C build/distribution JabRef.app && rm -R build/distribution/JabRef.app
+           archivePortable: brew install pigz && tar -c -C build/distribution JabRef.app | pigz --rsyncable > build/distribution/JabRef-portable_macos.tar.gz && rm -R build/distribution/JabRef.app
 
     runs-on: ${{ matrix.os }}
-    name: Deploy on ${{ matrix.displayName }}
+    name: Build for ${{ matrix.displayName }}
 
     steps:
-    - name: Checkout source
-      uses: actions/checkout@v2-beta
-      with:
-        fetch-depth: 0
-    - name: Fetch tags and master for GitVersion
-      run: |
-        git fetch --tags origin
-        git rev-parse --verify master
-        if (-not $?) {
-          git branch --force --create-reflog master origin/master
-        }
-      shell: pwsh
-    - name: Install GitVersion
-      uses: gittools/actions/setup-gitversion@v0.3
-      with:
-          versionSpec: '5.1.2'
-    - name: Run GitVersion
-      id: gitversion
-      uses: gittools/actions/execute-gitversion@v0.3
-    - name: Set up JDK
-      uses: actions/setup-java@v1
-      with:
-        java-version: 13
-    - uses: actions/cache@v1
-      with:
-        path: ~/.gradle/caches
-        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
-        restore-keys: |
-          ${{ runner.OS }}-gradle-${{ env.cache-name }}-
-          ${{ runner.OS }}-gradle-
-          ${{ runner.OS }}-
-    - uses: actions/cache@v1
-      name: Cache gradle wrapper
-      with:
-        path: ~/.gradle/wrapper
-        key: ${{ runner.os }}-gradle-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
-    - name: Download jpackage
-      # We need to download jpackage from https://jdk.java.net/jpackage/
-      run: |
-        import tarfile
-        import zipfile
-        import sys
-        if sys.version_info[0] >= 3:
-          from urllib.request import urlretrieve
-        else:
-          from urllib import urlretrieve
+      - name: Checkout source
+        uses: actions/checkout@v2-beta
+        with:
+          fetch-depth: 0
+      - name: Fetch tags and master for GitVersion
+        run: |
+          git fetch --tags origin
+          git rev-parse --verify master
+          if (-not $?) {
+            git branch --force --create-reflog master origin/master
+          }
+        shell: pwsh
+      - name: Install GitVersion
+        uses: gittools/actions/setup-gitversion@v0.3
+        with:
+            versionSpec: '5.1.2'
+      - name: Run GitVersion
+        id: gitversion
+        uses: gittools/actions/execute-gitversion@v0.3
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 13
+      - uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
```

Comment:
Why did you removed the restore-keys argument?
https://github.com/actions/cache/blob/master/examples.md#java---gradle is the official recommandation.

Reply:
I just triple-checked the **documentation** for `restore-keys`:

> `key` - An explicit key for restoring and **saving** the cache
> `restore-keys` - An ordered list of keys to use for restoring the cache if no cache hit occurred for key

I don't get how `${{ runner.os }}-gradle-` can ever be filled. Can you shed a light on this? Is the cache saved twice?

I don't have an issue when we re-fetch everything during a build. I think, we have more concerns with our cache size so that the concrete caches for build.gradle variants should be kept filled instead some global cache.

Reply:
@koppor  This one sheds light on it:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/caching-dependencies-to-speed-up-workflows#matching-a-cache-key 
>The cache action first searches for cache hits for key and restore-keys in the branch containing the workflow run. If there are no hits in the current branch, the cache action searches for key and restore-keys in the parent branch and upstream branches

Caching speeds up our workflow. And regarding cache size, I think only mac? was too large to store it, but that might change in the future